src/main.ts

```diff
@@ -56,76 +56,52 @@ function getInputs(): IActionInputs {
     const connectionString = core.getInput('connection-string', { required: true });
     const connectionConfig = new SqlConnectionConfig(connectionString);
 
-    // TODO: Deprecate server-name as input
-    let serverName = core.getInput('server-name', { required: false });
-    if ((!!serverName && !!connectionConfig.Config.server) && (serverName != connectionConfig.Config.server)) 
-        core.debug("'server-name' is conflicting with 'server' property specified in the connection string. 'server-name' will take precedence.");
-
-    // if serverName has not been specified, use the server name from the connection string
-    if (!serverName) serverName = connectionConfig.Config.server;
-
-    const additionalArguments = core.getInput('arguments');
-
-    let dacpacPackage = core.getInput('dacpac-package');
-    if (!!dacpacPackage) {
-        dacpacPackage = AzureSqlActionHelper.resolveFilePath(dacpacPackage);
-        if (path.extname(dacpacPackage).toLowerCase() !== Constants.dacpacExtension) {
-            throw new Error(`Invalid dacpac file path provided as input ${dacpacPackage}`);
-        }
-
-        if (!serverName) {
-            throw new Error(`Missing server name or address in the configuration.`);
-        }
-
-        return {
-            serverName: serverName,
-            connectionConfig: connectionConfig,
-            dacpacPackage: dacpacPackage,
-            sqlpackageAction: SqlPackageAction.Publish,
-            actionType: ActionType.DacpacAction,
-            additionalArguments: additionalArguments
-        } as IDacpacActionInputs;
-    }
-
-    let sqlFilePath = core.getInput('sql-file');
-    if (!!sqlFilePath) {
-        sqlFilePath = AzureSqlActionHelper.resolveFilePath(sqlFilePath);
-        if (path.extname(sqlFilePath).toLowerCase() !== '.sql') {
-            throw new Error(`Invalid sql file path provided as input ${sqlFilePath}`);
-        }
-
-        if (!serverName) {
-            throw new Error(`Missing server name or address in the configuration.`);
-        }
-
-        return {
-            serverName: serverName,
-            connectionConfig: connectionConfig,
-            sqlFile: sqlFilePath,
-            actionType: ActionType.SqlAction,
-            additionalArguments: additionalArguments
-        } as ISqlActionInputs;
+    let filePath = core.getInput('path', { required: true });
+    filePath = AzureSqlActionHelper.resolveFilePath(filePath);
+
+    // Optional inputs
+    const action = core.getInput('action');
```

Comment:
Why prompt for Action if only Publish is supported?

Reply:
for supporting more actions in the future
#106 will add some more